src/main/java/InfrastructureManager/Master.java

```diff
@@ -9,6 +12,9 @@
 
     private final CommandSet commandSet;
     private final ArrayList<Runner> runnerList;
+    private final List<EdgeNode> availableNodes;
+    private final Map<EdgeClient,EdgeNode> registeredClients;
```

Comment:
while it will functionally work the same, it is better to have a separate list for clients, and rename this to clientNodeMapping.